src/openrct2/paint/sprite/Paint.Sprite.cpp

```diff
@@ -32,13 +32,8 @@
  * Paint Quadrant
  *  rct2: 0x0069E8B0
  */
-void sprite_paint_setup(paint_session* session, const uint16_t x, const uint16_t y)
+void sprite_paint_setup(paint_session* session, const CoordsXY& pos)
 {
-    if ((x & 0xe000) | (y & 0xe000))
```

Comment:
Thinking about this some more and this should have been replaced with a 
```
    if (!map_is_location_valid(pos))
    {
        return;
    }```
As the setup function is expected to give invalid tiles when getting near the edges which are not required to be processed.